src/types/update.rs

```diff
@@ -113,31 +117,173 @@ pub enum UpdateKind {
     /// can_invite_users administrator right in the chat to receive these
     /// updates.
     ChatJoinRequest(ChatJoinRequest),
+
+    /// An error that happened during deserialization.
+    ///
+    /// This allows `teloxide` to continue working even if telegram adds a new
+    /// kind of updates.
+    Error(Value),
 }
 
-impl Update {
-    pub fn user(&self) -> Option<&User> {
-        match &self.kind {
-            UpdateKind::Message(m) => m.from(),
-            UpdateKind::EditedMessage(m) => m.from(),
-            UpdateKind::CallbackQuery(query) => Some(&query.from),
-            UpdateKind::ChosenInlineResult(chosen) => Some(&chosen.from),
-            UpdateKind::InlineQuery(query) => Some(&query.from),
-            UpdateKind::ShippingQuery(query) => Some(&query.from),
-            UpdateKind::PreCheckoutQuery(query) => Some(&query.from),
-            UpdateKind::PollAnswer(answer) => Some(&answer.user),
-            _ => None,
+impl<'de> Deserialize<'de> for UpdateKind {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct Visitor;
+
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = UpdateKind;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a map")
+            }
+
+            fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+            where
+                A: MapAccess<'de>,
+            {
+                let mut tmp = None;
+
+                // Try to deserialize a borrowed-str key, or else try deserializing an owned
+                // string key
+                let k = map.next_key::<&str>().or_else(|_| {
+                    map.next_key::<String>().map(|k| {
+                        tmp = k;
+                        tmp.as_deref()
+                    })
+                });
+
+                if let Ok(Some(k)) = k {
+                    let res = match k {
+                        "message" => dbg!(map
```

Comment:
Is that `dbg!()` intended or forgot to remove?

Reply:
622b0d9232bc52a41c51af7e1be8ae439e8cafd5